src/runtime/hexagon/hexagon_device_api.cc

```diff
@@ -90,18 +90,20 @@ void* HexagonDeviceAPI::AllocDataSpace(Device dev, int ndim, const int64_t* shap
 
   const size_t typesize = (dtype.bits / 8) * dtype.lanes;
 
+  HexagonBufferManager* mgr = runtime_hexbuffs ? runtime_hexbuffs.get() : &hexbuffs;
```

Comment:
It would be best if we could fully find and fix the "init time" uses of `hexbuffs` before merging this PR.

If that's not possible, then `mgr` should be a member of HexagonDeviceAPI which is initialized with `&hexbuffs` and then set to `std::make_unique<HexagonBufferManager>()` in "acquire" to avoid all this trinary switching at runtime.

Reply:
I was keeping the check for both managers in free.  However, now that I'm calling Acquire/Release in the session object directly, I can switch to having a "current" manager, and remove those trinary statements.